default/python/AI/PolicyAI.py

```diff
@@ -0,0 +1,235 @@
+from EnumsAI import PriorityType
+
+propaganda = "PLC_PROPAGANDA"
+algo_research = "PLC_ALGORITHMIC_RESEARCH"
+moderation = "PLC_MODERATION"
+industrialism = "PLC_INDUSTRIALISM"
+technocracy = "PLC_TECHNOCRACY"
+centralization = "PLC_CENTRALIZATION"
+bureaucracy = "PLC_BUREAUCRACY"
+infra1 = "PLC_PLANETARY_INFRA"
+infra2 = "PLC_SYSTEM_INFRA"
+infra3 = "PLC_INTERSTELLAR_INFRA"
+basics = [
+    propaganda,
+    algo_research,
+    infra1,
+    # no use for the extra slot yet and they are expensive
+    # infra2,
+    # infra3,
+]
+
+
+class PolicyManager:
+    """Policy Manager for one round"""
+
+    def __init__(self):
+        # resourceAvailable includes this turns production, but that is wrong for influence
+        self.empire = fo.getEmpire()
+        self.universe = fo.getUniverse()
+        self.ip = self.empire.resourceAvailable(fo.resourceType.influence) - self.get_infl_prod()
+        self.adopted = set(self.empire.adoptedPolicies)
+        self.adoptable = self.get_adoptable()
+        empire_owned_planet_ids = PlanetUtilsAI.get_owned_planets_by_empire(self.universe.planetIDs)
+        self.planet_ids = PlanetUtilsAI.get_populated_planet_ids(empire_owned_planet_ids)
+        self.num_populated = len(self.planet_ids)
+        self.num_outposts = len(empire_owned_planet_ids) - self.num_populated
+        self.aistate = get_aistate()
+
+    def generate_orders(self):
+        """generate policy orders"""
+        debug(
+            "Start turn %d IP: %.2f + %.2f, adoptable: %s, adopted: %s",
+            fo.currentTurn(),
+            self.ip,
+            self.get_infl_prod(),
+            str(self.adoptable),
+            str(self.adopted),
+        )
+        for policy in basics:
+            if policy in self.adoptable:
+                self.adopt(policy)
+        # we need the extra slots first
+        if infra1 in self.adopted:
+            if bureaucracy in self.adopted:
+                max_turn = 19  # AI doesn't build Regional Admins yet (TBD count, palace may be gone, too)
+                if fo.currentTurn() >= self.empire.turnsPoliciesAdopted[bureaucracy] + max_turn:
+                    self.deadopt(bureaucracy)
+                    self.try_adopt_centralization()
+            else:
+                self.try_adopt_bureaucracy()
+            self.check_deadopt_centralization()
+            self.techno_or_industry()
+        new_production = self.get_infl_prod(True)
+        debug("End of turn IP: %.2f + %.2f", self.ip, new_production)
+        # priority shouldn't only be based on current value
+        forecast = self.ip + 3 * new_production
+        if infra1 in self.adopted and bureaucracy not in self.adopted and centralization not in self.adopted:
+            ccost = fo.getPolicy(centralization).adoptionCost()
+            bcost = fo.getPolicy(bureaucracy).adoptionCost()
+            forecast -= self.num_populated + ccost + bcost
+        if forecast < 40:
+            prio = 60 - forecast
```

Comment:
When I get the not word in the code, you have to either guess or look into the code.   So let's just use full names.

We already have some conventions for names like `pid` (planet id), but I try to avoid adding others.

Reply:
I think I've resolved all your comments and did, but this one I do not understand.

Reply:
`prio` is not a word.  So I need to guess it. Looks like a priority, but it's not a good match with the word forecast.  So I need to break reading the code and do a forward lookup on what this variable is. 

This decreases the readability.   

It does not slow down me a lot, but fixing it is also very easy.  By fixing I mean not only change this variable, but also prevent other such cases in the future.